src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs

```diff
@@ -4564,6 +4564,32 @@ bool M<T>(ref T x)
 ignoreTrivia: false);
         }
 
+        [WorkItem(16819, "https://github.com/dotnet/roslyn/issues/16819")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task InlineVariableDoesNotAddsDuplicateCast()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M()
+    {
+        var [||]o = (Exception)null;
+        Console.Write(o == new Exception());
+    }
+}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        Console.Write((Exception)null == new Exception());
+    }
+}");
+        }
+
```

Comment:
This should probably go at the end of the file rather than between two tests that are related to one another (`InlineTemporary_RefParams` and `InlineTemporary_OutParams`).